airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -210,21 +238,25 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Ite
 
         stream_state = stream_state or {}
         channels_stream = Channels(authenticator=self.authenticator)
-        if "start_ts" in stream_state:
+        if self.cursor_field in stream_state:
             # Since new messages can be posted to threads continuously after the parent message has been posted, we get messages from the latest date
             # found in the state minus 7 days to pick up any new messages in threads.
             # If there is state always use lookback
-            messages_start_date = pendulum.from_timestamp(stream_state.get("start_ts")).subtract(**self.messages_lookback_window)
+            try:
+                messages_start_date = pendulum.from_timestamp(stream_state[self.cursor_field]).subtract(**self.messages_lookback_window)
+            except TypeError:  # during second read we will have ts as a string date
+                messages_start_date = pendulum.parse(stream_state[self.cursor_field]).subtract(**self.messages_lookback_window)
```

Comment:
I don't think this works without changes after we replace dict with Duration

Reply:
I run the tests and they are OK

Reply:
so you know now that tests are not good enough

Reply:
fixed this